src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1876,10 +1877,10 @@ private BoundExpression BindCast(CastExpressionSyntax node, DiagnosticBag diagno
                 return BindExplicitNullableCastFromNonNullable(node, operand, targetType, diagnostics);
             }
 
-            return BindCastCore(node, operand, targetType, wasCompilerGenerated: operand.WasCompilerGenerated, diagnostics: diagnostics);
+            return BindCastCore(node, operand, targetType, wasCompilerGenerated: operand.WasCompilerGenerated, diagnostics: diagnostics, isNullable: targetTypeWithNullability.IsNullable == true);
```

Comment:
Nit: perhaps break long lines (1880, 1883, 1905 would be candidates imo)

Reply:
I agree with breaking long lines but I'd like to minimize the changes from `master`.